tools/markdown_lint/linter.py

```
@@ -0,0 +1,816 @@
+                    in_html_comment = False  # Single line comment
+                    prev_line = line
+                    prev_line_blank = False
+                    continue
+
+                # Check line length
+                self._check_line_length(report, i, line)
+
+                # Check for trailing whitespace
+                if self.config["trim_trailing_whitespace"] and line.rstrip() != line:
+                    self._add_issue(
+                        report, i, "Trim trailing whitespace", "MD009", fix=lambda l: l.rstrip()
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)